sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/ETag.java

```diff
@@ -68,8 +68,8 @@ public static ETag fromString(String eTag) {
         if (!endsWithQuote || (!startsWithQuote && !startsWithWeakETagPrefix)) {
             throw LOGGER.atError()
                 .addKeyValue("ETag", eTag)
-                .log(new IllegalArgumentException(
-                    "The ETag should be null, '*', be wrapped in quotes, or be wrapped " + "in quotes prefixed by W/"));
+                .logThrowable(new IllegalArgumentException(
```

Comment:
if `logThrowable` follows `setThrowable` then the throwable in the last call take precedence, right ?

Reply:
yes, the `throw logger.atError().setThrowable(ex1).logThrowable(ex2)` would ignore `ex1`

Reply:
I am not sure we need 2 ways to log an exception if only one is going to be used. I am in favor of dropping one of the two methods. Either we keep `setThrowable` or `logThrowable`. My preference would be to keep `setThrowable` as we don't need to add multiple overloads for logThrowable(ex), logThrowable(string, ex), logThrowable(supplier, ex)